site/website/blog/2020-10-06-enabling-full-text-search.md

```diff
@@ -0,0 +1,492 @@
+---
+title: Enabling Full-text Search with Change Data Capture in a Legacy Application
+description: Debezium based Change Data Capture sources for Hazelcast Jet
+author: František Hartman
+authorURL: https://www.linkedin.com/in/frantisek-hartman/
+authorImageURL: https://i.stack.imgur.com/3X7wE.png 
+---
+
+In this post, we will take a legacy application and improve its search
+functionality. We will do this with only a few changes to the
+application itself - the technique called change data capture  (CDC)
+allows us to listen for changes in a database and react on these
+changes&nbsp;- write to a search index in our case. We will also enrich
+the data before we write it to the search index with a natural language
+processing (NLP) step that extracts keywords from a text description.
+
+We might want to use an external search index for various reasons,
+especially:
+
+* Providing more feature-rich functionality than full-text search in a
+relational database.
+
+* It allows us to scale the search independently of the database.
+
+* Add more functionality (keyword extraction) to the application without
+modifying it - making changes to a legacy application is sometimes
+risky, or time-consuming, so taking this approach might be faster,
+therefore cheaper.  
+
+In this tutorial we will do the following:
+
+* Use Hazelcast Jet - an open-source stream processing system - and it’s
+CDC module to read changes made to the application database for
+further processing in a Jet pipeline
+
+* In the Jet pipeline, we will enrich the data with an NLP mapping step
```

Comment:
maybe write out natural language processing, I already forgot what the acronym meant, by the time I got here